src/Shared/Components/GenericInfoCard/GenericInfoCardListing.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024. Devtron Inc.
+ */
+
+import { useMemo } from 'react'
+
+import emptyList from '@Images/empty-create.png'
+import ErrorScreenManager from '@Common/ErrorScreenManager'
+import { GenericEmptyState, GenericFilterEmptyState } from '@Common/index'
+
+import GenericInfoCard from './GenericInfoCard.component'
+import { GenericInfoListSkeleton } from './GenericInfoListSkeleton'
+import { GenericInfoCardListingProps } from './types'
+
+export const GenericInfoCardListing = ({
+    isLoading,
+    error,
+    list,
+    searchKey,
+    reloadList,
+    borderVariant,
+    handleClearFilters,
+    emptyStateConfig,
+}: GenericInfoCardListingProps) => {
+    const filteredList = useMemo(() => {
+        if (!searchKey || error) {
```

Comment:
We should use sanitized list, since on loading this might by null
```suggestion
        const sanitizedList = list || []
        if (!searchKey || error) {
```